Open Telemetry

Xaelias opened the pull request:
This PR implements otel for http and thrift in bp.py.
Because of unsolvable conflicting dependencies, this also removes python 3.7.

Xaelias committed:
feat: adhoc: Update pre-commit hooks and Dockerfile for local dev

Xaelias committed:
feat: adhoc: Install opentelemetry pacakges, create Reddit trace propagator and log trace IDs

Xaelias committed:
feat: adhoc: Implement opentelemetry for thrift client and server

Xaelias committed:
feat: adhoc: Instrument pyramid/request with Opentelemetry

trevorriles commented:
@chriskuehl you mentioned that there is some changes that have made it to develop but not been released yet. Would it be better for us to target a release branch?

chriskuehl commented on .github/workflows/python-package.yaml:
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9]
```
Looks like we duplicated some of the Python 3.7 dropping work; I made similar changes in #823 (isolated in the first commit).

Maybe it makes sense for us to pull that into a separate PR? TODO for me.

chriskuehl replied:
Extracted to https://github.com/reddit/baseplate.py/pull/824

chriskuehl commented on Dockerfile:
```diff
@@ -12,6 +12,7 @@ WORKDIR /src
 ENV CASS_DRIVER_NO_EXTENSIONS theytaketoolongtobuild
 
 COPY requirements*.txt ./
+RUN apt-get -y update && apt-get -y install librdkafka-dev && rm -rf /var/lib/apt/lists/*
```
Do you recall what package needed this? I wonder if there's a possibility of choosing a version that has manylinux wheels uploaded to avoid needing to compile it ourselves at install time.

Xaelias replied:
I am not 100% sure anymore but I think it had smth to do with the apple silicon MBPs.

chriskuehl replied:
Ah, interesting. I actually changed it to use x86_64 emulation in my PR (https://github.com/reddit/baseplate.py/pull/823/commits/97ff89d5bca69405e016d799c742fcb5ec697c6a) since there was no Thrift compiler image for ARM and it was failing on my MacBook trying to build a native arch version of the image. This is probably fine to keep as-is for now, though I'm surprised you didn't run into the same Thrift issue I did (maybe I'm doing something wrong?).

Xaelias replied:
```DOCKER_DEFAULT_PLATFORM=linux/amd64 docker-compose build```
worked locally for me

Xaelias replied:
so two different paths to achieve somewhat similar results :-D

chriskuehl commented on baseplate/clients/requests.py:
```diff
@@ -26,6 +27,9 @@
 from baseplate.lib.prometheus_metrics import getHTTPSuccessLabel
 
 
+RequestsInstrumentor().instrument()
```
It looks like the way these work is essentially to patch these libraries to inject tracing logic. I don't think that's inherently a problem but it's definitely a little sneaky, especially to do as an import-time side effect.

In general I think it's best if importing code doesn't have any side effects. This matches people's intuition about what importing does, reduces weird surprises (e.g. adding a seemingly-unrelated import suddenly changes behavior), is faster, and plays better with linters and any other tools that need to import code. It also makes it possible to access any configuration the app provides so that you can adjust the instrumentation settings if needed (e.g. if we wanted to add a way to toggle off instrumentation, or make it possible to exempt certain routes).

Do you think there's a way we can move this out of the import path and instead have it happen at startup time?

Xaelias replied:
Honestly I don't know. I think it would make testing much more complex since you'd need to basically load the entire app to test anything.
It's also how otel says you should be using these auto-instrumenter.

KTAtkinson replied:
I agree with Chris that this smells a little funny. I don't see anything in their API docs that suggests that this needs to be instrumented the top level like this. Can you share the link to the instructions/example you are following?

Xaelias replied:
Can you check the commit I just pushed? Is that closer to what y'all had in mind? (we change names etc. I just wanted to get smth out there for you to be able to look at)

trevorriles replied:
I think @Xaelias 's change works for me what are your thoughts @KTAtkinson & @chriskuehl ?

chriskuehl commented on baseplate/clients/thrift.py:
```diff
@@ -191,6 +201,15 @@ def __init__(
+        try:
+            self.otel_peer_name = socket.getfqdn()
+            self.otel_peer_ip = socket.gethostbyname(self.otel_peer_name)
+        except Exception:
```
```suggestion
        except socket.gaierror:
```

chriskuehl commented on baseplate/clients/thrift.py:
```diff
@@ -223,119 +260,150 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
+                            otelspan.set_status(status.Status(status.StatusCode.OK))
+                            span.finish()
+                            raise
+                        except:  # noqa: E722
```
```suggestion
                        except BaseException:
```

Explicitly writing `except BaseException:` is the way to avoid the linter error when that's your goal.

chriskuehl commented on baseplate/frameworks/pyramid/__init__.py:
```diff
@@ -35,10 +37,12 @@
 from baseplate.thrift.ttypes import IsHealthyProbe
 
 
+PyramidInstrumentor().instrument()
```
Same question as before about instrumenting as an import-time side effect.

chriskuehl commented on baseplate/frameworks/pyramid/__init__.py:
```diff
@@ -64,10 +68,15 @@ def __next__(self) -> bytes:
             raise
-        except:  # noqa: E722
-            self.span.finish(exc_info=sys.exc_info())
+        except Exception as e:  # noqa: E722
```
This is a behavior change (bare except catches all exceptions, `except Exception` catches only exceptions that inherit from Exception. So, this no longer catches exceptions like `KeyboardInterrupt`. Is that OK? If you want to keep the old behavior, you can use `except BaseException as e` instead, which is equivalent to the bare except.

You should be able to drop the `noqa` here now that you're not using a bare except.

trevorriles replied:
In this case I think we are ok using `Exception`. The opentelemetry lib want's an `Exception` for it's `record_exception` function. As in all these cases we just end the span and re-raise, I think it's fine if we miss ending the span for exceptions that inherit from `BaseException` but not `Exception`.

chriskuehl commented on baseplate/frameworks/pyramid/__init__.py:
```diff
@@ -129,15 +138,21 @@ def baseplate_tween(request: Request) -> Response:
             if request.span:
                 request.span.set_tag("http.response_length", response.content_length)
-        except:  # noqa: E722
+        except Exception as e:  # noqa: E722
```
Ditto

chriskuehl commented on baseplate/frameworks/thrift/__init__.py:
```diff
@@ -80,87 +124,149 @@ def call_with_context(*args: Any, **kwargs: Any) -> Any:
+                        # Set as OK as this is an expected exception
+                        otelspan.set_status(trace.status.Status(trace.status.StatusCode.OK))
+                        raise
+                    except Exception as exc:  # noqa: E722
```
Probably can drop noqa, same comments as before.

chriskuehl commented on baseplate/lib/propagator_redditb3.py:
```diff
@@ -0,0 +1,158 @@
+    SPAN_ID_KEY = "X-Span"
+    SAMPLED_KEY = "X-Sampled"
+    FLAGS_KEY = "X-Flags"
+    _SAMPLE_PROPAGATE_VALUES = {"1", "True", "true", "d"}
```
```suggestion
    _SAMPLE_PROPAGATE_VALUES = frozenset(("1", "True", "true", "d"))
```

chriskuehl commented on baseplate/lib/propagator_redditb3.py:
```diff
@@ -0,0 +1,158 @@
+    FLAGS_KEY = "X-Flags"
+    _SAMPLE_PROPAGATE_VALUES = {"1", "True", "true", "d"}
+    # Although Reddit B3 uses 64bit TraceId's (16 char) we will accept 32 and truncate only when writing client headers.
+    _trace_id_regex = re_compile(r"[\da-fA-F]{16}|[\da-fA-F]{32}")
```
Since there's no anchor at the end, I think it will always just select the first 16 chars even for a longer ID?

```python
>>> re.match(r"[\da-fA-F]{16}|[\da-fA-F]{32}", "aaaabbbbccccddddaaaabbbbccccdddd").group(0)
'aaaabbbbccccdddd'
```

It also allows other character lengths as long as they're at least 16. Not sure exactly what the intention of the regex is here but given that the `|[\da-fA-F]{32}` part isn't doing anything, this seems unexpected?

trevorriles replied:
I think your correct here, I lifted this from upstream here and modified it for Reddit's format.
https://github.com/open-telemetry/opentelemetry-python/blob/d6321d627cd0fc9345a689a9a37cf4cb9ea44ba0/propagator/opentelemetry-propagator-b3/src/opentelemetry/propagators/b3/__init__.py#L46-L47

Let me dig in a bit more to make sure we get the best behavior.

trevorriles replied:
Ahh it uses the [re.fullmatch](https://docs.python.org/3/library/re.html#re.fullmatch) function, which if I read it correctly means it will check the second option.

trevorriles replied:
Either way this is incorrect as the baseplate spec uses base10 encoded integers... I'll get a fix together.

trevorriles replied:
Got the fix in to look for base10 integers wrapped in a string.

chriskuehl reviewed (commented):
First pass review -- it would be a good idea to have someone more familiar with the tracing bits also take a look.

Xaelias committed:
trying to move the auto instrumentation around as requested

trevorriles commented:
I've got develop merged in and all the tests passing. I'll begin addressing the other PR comments next.

One thing to note is that opentelemetry-python doesn't officially support 3.12 yet, so we may have to wait for that or restrict baseplate to 3.11 for the time being.

Commit:
Addressed some PR comments

Commit:
Change these back from BaseException to Exception

I think we are fine here since we re-raise the exception anyways, if you miss ending a span for exceptions that are not included in  I think that is OK.

Commit:
Switch thrift back to  and record the exception to otel.

Commit:
Don't explicitly record the exception as we enabled the auto exception recording option when creating the span

Commit:
Update propagator to correctly follow baseplate spec using base10 integers

This also shifts to the assumption that we do not truncate the 128 bit trace id but send it as is.

Commit:
Apply linting fixes

Commit:
fix

Commit:
Fix pyramid tests to reflect the baseplate spec's use of base10 trace/span ids.

Commit:
Update how we configure propagators so that we support the different headers between thrift and http

The difference here is that our HTTP headers have the 'X-' prefix and Thrift does not according to the baseplate spec.

Commit:
Fix formatting and linting

trevorriles commented:
@chriskuehl I'm curious what your thoughts are on the lack of 3.12 support at the moment?

https://github.com/open-telemetry/opentelemetry-python/issues/3617

chriskuehl commented:
> @chriskuehl I'm curious what your thoughts are on the lack of 3.12 support at the moment?

I don't think we can merge this until 3.12 support is available, since we already have services which have migrated to 3.12.

chriskuehl commented on Dockerfile:
```diff
@@ -7,6 +7,7 @@ COPY --from=thrift /usr/local/bin/thrift /usr/local/bin/thrift
 WORKDIR /src
 
 COPY requirements*.txt ./
+RUN apt-get -y update && apt-get -y install librdkafka-dev && rm -rf /var/lib/apt/lists/*
```
Can we move this new apt line above the COPY so that we don't need to reinstall apt packages every time a requirement changes? Maybe it can go directly under the `FROM python:3.12` line?

trevorriles replied:
Good call, I'll move this up.

chriskuehl commented on baseplate/clients/thrift.py:
```diff
@@ -223,119 +264,150 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
+                    elif pool_addr is not None:
+                        otel_attributes[SpanAttributes.NET_PEER_IP] = pool_addr.host
+                        otel_attributes[SpanAttributes.NET_PEER_PORT] = pool_addr.port
+                    if otel_attributes.get(SpanAttributes.NET_PEER_IP) in ["127.0.0.1", "::1"]:
+                        otel_attributes[SpanAttributes.NET_PEER_NAME] = "localhost"
```
When does localhost traffic like this happen? For the IPv4 address I guess in theory it could be all of 127.0.0.0/8 depending on how this is used? Mostly wondering because it's not uncommon to see e.g. `127.0.1.1` used in default Linux setups in `/etc/hosts`.

trevorriles replied:
Honestly I think the effort here is to just provide a default `NET_PEER_NAME` for some common scenarios. It's kind of a convenience if you are filtering or grouping by hostnames I guess so that you get a value instead of empty.

chriskuehl commented on baseplate/lib/propagator_redditb3_http.py:
```diff
@@ -0,0 +1,157 @@
+    FLAGS_KEY = "X-Flags"
+    _SAMPLE_PROPAGATE_VALUES = frozenset({"1", "True", "true", "d"})
+    # Reddit B3 trace and span id's are 64bit integers encoded as decimal
+    _id_regex = re_compile(r"[\d]+")
```
```suggestion
    _id_regex = re_compile(r"\d+")
```

chriskuehl commented on baseplate/lib/propagator_redditb3_thrift.py:
```diff
@@ -0,0 +1,157 @@
+    FLAGS_KEY = "Flags"
+    _SAMPLE_PROPAGATE_VALUES = frozenset({"1", "True", "true", "d"})
+    # Reddit B3 trace and span id's are 64bit integers encoded as decimal
+    _id_regex = re_compile(r"[\d]+")
```
```suggestion
    _id_regex = re_compile(r"\d+")
```

chriskuehl commented on baseplate/server/__init__.py:
```diff
@@ -190,6 +214,95 @@ def configure_logging(config: Configuration, debug: bool) -> None:
+                )
+            # Load insecure option from tracing config.
+            insecure: bool
+            if "insecure" in config.tracing:
+                if config.tracing["insecure"] == "false":
+                    insecure = False
+            else:
+                # Default to true. At the time of implementation our current tracing pipeline is not setup for TLS.
+                insecure = True
```
```suggestion
            if config.tracing.get("insecure") == "false":
                insecure = False
            else:
                # Default to true. At the time of implementation our current tracing pipeline is not setup for TLS.
                insecure = True
```

chriskuehl commented on tests/integration/otel_thrift_tests.py:
```diff
@@ -0,0 +1,963 @@
+        if not metrics:
+            return
+
+        try:
+            metrics.get_active_requests_metric().clear()
+            metrics.get_latency_seconds_metric().clear()
+            metrics.get_requests_total_metric().clear()
+        except Exception:
+            pass
```
How does this exception happen? Is it possible to catch a specific error?

trevorriles replied:
I'll look into the exceptions thrown here and clean this up.

trevorriles replied:
This method wasn't called so I just removed it.

I also removed it from the `thrift_tests.py` file as well as that's where we lifted most of these tests.

chriskuehl reviewed (approved):
This looks reasonable but given the size it's difficult to catch any bugs during review. I think we should discuss how we want to test this before the final release.

Let's also get a review from someone more familiar with our tracing setup before merging, since I'm not really qualified to review that beyond basic logic checking.

trevorriles committed:
regex fix

Co-authored-by: Chris Kuehl <chris.kuehl@reddit.com>

trevorriles committed:
regex fix x2

Co-authored-by: Chris Kuehl <chris.kuehl@reddit.com>

trevorriles committed:
Simplify conditional logic in server init

Co-authored-by: Chris Kuehl <chris.kuehl@reddit.com>

trevorriles commented:
> This looks reasonable but given the size it's difficult to catch any bugs during review. I think we should discuss how we want to test this before the final release.

Fully on board here, here are two testing steps I want to take before we cut an official release:
- [ ] Additional testing in snoodev to verify context propagation general functionality
- [ ] Cut some form of alpha release and work with teams that are still < 3.12. This will make it easier for them to test in snoodev and possibly even some canary testing.

With the 3.12 limitation I'm interested in your thoughts around maintaining an otel branch/release that will be limited to 3.11 and below until we can get the upstream issues resolved?

> Let's also get a review from someone more familiar with our tracing setup before merging, since I'm not really qualified to review that beyond basic logic checking.

I'm sure we can get @Xaelias to give this another pass, since I've taken this over. Not sure if you can approve your own PR though. Ultimately we are the two people who have context to the tracing work here. Pete Loaf has done most of the golang work, but I'm not sure if it makes sense to have him review here as well or not?

trevorriles commented on baseplate/server/__init__.py:
```diff
@@ -190,6 +214,94 @@ def configure_logging(config: Configuration, debug: bool) -> None:
+            )
+
+
+def otel_tracing_autoinstrumentation() -> None:
+    PyramidInstrumentor().instrument()
+    RequestsInstrumentor().instrument()
```
I need to look at moving these back.

This created an infinite loop somehow related to the import timing of the `ssl` lib and our gevent `patch_all()` call in `baseplate-serve`. 

Honestly I don't 100% know why yet, still digging.

trevorriles replied:
Or an alternative that is not this 😓

trevorriles committed:
Move instrumentation back to modules instead of __init__ (#846)

Co-authored-by: Trevor Riles <trevor@reddit.com>

trevorriles committed:
Trevorriles/temp (#910)

Add support for thrift header propagation.

Commit:
Rebase fixes

Commit:
Update cached_property wrapper

This change is based on the upstream pyramid reify code we orginally lifted this from.

Commit:
Add back in baseplate thrift errorcode fix

Commit:
Fix indentation error made during rebase

Commit:
Set doc pyproject black language level to match top-level project

Commit:
Fix mypy type that was missed during rebase

Commit:
Remove reset_metrics method in tests that did not call it

trevorriles committed:
Move baseplate.py sampling and configuration to be more in line with … (#920)

* Move baseplate.py sampling and configuration to be more in line with baseplate.go implementation

---------

Co-authored-by: Trevor Riles <trevor@reddit.com>

Commit:
Merge branch 'develop' of github.com:reddit/baseplate.py into aleksei.lesieur/opentelemetry2

Commit:
make fmt

trevorriles commented on baseplate/server/__init__.py:
```diff
@@ -190,6 +206,54 @@ def configure_logging(config: Configuration, debug: bool) -> None:
+
+def configure_tracing() -> None:
+    logger.info("Entering configure tracing function")
+    sample_rps = 10
+    try:
+        sample_rps = DefaultFromEnv(Integer, "BASEPLATE_TRACE_SAMPLER_RPS", 10)
+    except ValueError:
+        logger.warning(
+            "Invalid BASEPLATE_TRACE_SAMPLER_RPS, falling back to the default of 10 RPS."
+        )
```
@chriskuehl I'm trying to use `DefaultFromEnv` as suggested but I think I'm misunderstanding how this is supposed to work. Does this also require implementing these options in the ini based config to get it to work?

dependabot[bot] committed:
Bump lxml from 5.2.1 to 5.2.2 (#921)

Bumps [lxml](https://github.com/lxml/lxml) from 5.2.1 to 5.2.2.
- [Release notes](https://github.com/lxml/lxml/releases)
- [Changelog](https://github.com/lxml/lxml/blob/master/CHANGES.txt)
- [Commits](https://github.com/lxml/lxml/compare/lxml-5.2.1...lxml-5.2.2)

---
updated-dependencies:
- dependency-name: lxml
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump boto3 from 1.34.95 to 1.34.117 (#922)

Bumps [boto3](https://github.com/boto/boto3) from 1.34.95 to 1.34.117.
- [Release notes](https://github.com/boto/boto3/releases)
- [Changelog](https://github.com/boto/boto3/blob/develop/CHANGELOG.rst)
- [Commits](https://github.com/boto/boto3/compare/1.34.95...1.34.117)

---
updated-dependencies:
- dependency-name: boto3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 4.1.4 to 4.1.6 (#927)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.4 to 4.1.6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/0ad4b8fadaa221de15dcec353f45205ec38ea70b...a5ac7e51b41094c92402da3b24376905380afc29)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

redloaf commented on baseplate/frameworks/thrift/__init__.py:
```diff
@@ -64,11 +83,40 @@ def __init__(
+                    self.logger.info("Unable to decode header %s, ignoring." % k.decode())
+
+            ctx = propagator.extract(header_dict)
+            logger.debug("Extracted trace headers. [ctx=%s, header_dict=%s]" % (ctx, header_dict))
```
🔕 I believe this formats the log message even if debug logging is disabled.
```suggestion
            logger.debug("Extracted trace headers. [ctx=%s, header_dict=%s]", ctx, header_dict)
```

chriskuehl committed:
v2.7.0

trevorriles committed:
Update baseplate/frameworks/thrift/__init__.py

Co-authored-by: redloaf <99754113+redloaf@users.noreply.github.com>

dependabot[bot] committed:
Bump urllib3 from 1.26.18 to 1.26.19 (#931)

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.18 to 1.26.19.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/1.26.19/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.18...1.26.19)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump typing-extensions from 4.12.0 to 4.12.2 (#939)

Bumps [typing-extensions](https://github.com/python/typing_extensions) from 4.12.0 to 4.12.2.
- [Release notes](https://github.com/python/typing_extensions/releases)
- [Changelog](https://github.com/python/typing_extensions/blob/main/CHANGELOG.md)
- [Commits](https://github.com/python/typing_extensions/compare/4.12.0...4.12.2)

---
updated-dependencies:
- dependency-name: typing-extensions
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump moto from 5.0.9 to 5.0.10 (#938)

Bumps [moto](https://github.com/getmoto/moto) from 5.0.9 to 5.0.10.
- [Release notes](https://github.com/getmoto/moto/releases)
- [Changelog](https://github.com/getmoto/moto/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getmoto/moto/compare/5.0.9...5.0.10)

---
updated-dependencies:
- dependency-name: moto
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump pylint from 3.2.2 to 3.2.5 (#937)

Bumps [pylint](https://github.com/pylint-dev/pylint) from 3.2.2 to 3.2.5.
- [Release notes](https://github.com/pylint-dev/pylint/releases)
- [Commits](https://github.com/pylint-dev/pylint/compare/v3.2.2...v3.2.5)

---
updated-dependencies:
- dependency-name: pylint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump flake8 from 7.0.0 to 7.1.0 (#936)

Bumps [flake8](https://github.com/pycqa/flake8) from 7.0.0 to 7.1.0.
- [Commits](https://github.com/pycqa/flake8/compare/7.0.0...7.1.0)

---
updated-dependencies:
- dependency-name: flake8
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump types-setuptools from 70.0.0.20240524 to 70.1.0.20240627 (#935)

Bumps [types-setuptools](https://github.com/python/typeshed) from 70.0.0.20240524 to 70.1.0.20240627.
- [Commits](https://github.com/python/typeshed/commits)

---
updated-dependencies:
- dependency-name: types-setuptools
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump boto3 from 1.34.117 to 1.34.136 (#932)

Bumps [boto3](https://github.com/boto/boto3) from 1.34.117 to 1.34.136.
- [Release notes](https://github.com/boto/boto3/releases)
- [Commits](https://github.com/boto/boto3/compare/1.34.117...1.34.136)

---
updated-dependencies:
- dependency-name: boto3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump certifi from 2024.2.2 to 2024.7.4 (#944)

Bumps [certifi](https://github.com/certifi/python-certifi) from 2024.2.2 to 2024.7.4.
- [Commits](https://github.com/certifi/python-certifi/compare/2024.02.02...2024.07.04)

---
updated-dependencies:
- dependency-name: certifi
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Remove broken environment variable based configuration. Sampling config will come with a future release keeping us aligned with baseplate.go v2

Commit:
Remove unused imports
